src/utils/database.utils.js

```diff
@@ -0,0 +1,71 @@
+import { Op } from 'sequelize';
+
+import models from '../db/models';
+
+const { Article, User, Tag } = models;
+
+const articleAttributes = [
+  'title',
+  'description',
+  'body',
+  'status',
+  'readTime',
+];
+
+const userAttributes = ['username', 'firstName', 'lastName', 'image'];
+
+const tagAttributes = ['name'];
+
+export const articleFilter = async keyword => {
+  return await Article.findAll({
+    attributes: articleAttributes,
+    where: {
+      title: {
+        [Op.iLike]: `%${keyword}%`,
+      },
+    },
+  });
+};
+
+export const authorsFilter = async keyword => {
+  return await User.findAll({
+    attributes: userAttributes,
+    where: {
+      [Op.or]: [
+        {
+          firstName: {
+            [Op.iLike]: `%${keyword}%`,
+          },
+        },
+        {
+          lastName: {
+            [Op.iLike]: `%${keyword}%`,
+          },
+        },
+        {
+          username: {
+            [Op.iLike]: `%${keyword}%`,
+          },
+        },
+      ],
+    },
+  });
+};
+
+export const tagsFilter = async keyword => {
+  return await Tag.findAll({
+    attributes: tagAttributes,
+    where: {
+      name: {
+        [Op.iLike]: `%${keyword}%`,
+      },
+    },
+    include: [
```

Comment:
we can remove this include and just return the list of tags

Reply:
We have removed it now